docs/ldml/tr35.md

```diff
@@ -2861,38 +2861,52 @@ Element content whose display may be affected in this way should include an expl
+A UnicodeSet may be cited in specifications outside of the domain of LDML.
+In such a case, that specification may specify a subset or superset of the syntax provided here.
 
-The following provides EBNF syntax for a UnicodeSet:
+##### UnicodeSet syntax #####
 
 | Symbol         | Expression                                                     | Examples                                |
 | -------------- | -------------------------------------------------------------- | --------------------------------------- |
-| `root`         | <pre>= prop<br/>\| '[-]'<br/>\| '[' [\\-\\^]? s seq+ ']'</pre> | \\p{x=y},<br/>[abc]                     |
-| `seq`          | <pre>= root (s [\\&\\-] s root)* s<br/>\| range s</pre>        | [abc]-[cde], a                          |
-| `range`        | <pre>= char ('-' char)?<br/>\| '{' (s char)+ s '}'</pre>       | a, a-c, \{abc}                          |
-| `prop`         | <pre>= '\\' [pP] '{' propName ([≠=] s value1+)? '}'<br/>\| '[:' '^'? propName ([≠=] s value2+)? ':]'</pre> | \\p\{x=y}, [:x=y:]<br/> |
-| `propName`     | <pre>= s [A-Za-z0-9] [A-Za-z0-9_\\x20]* s</pre>                | General_Category,<br/>General Category  |
-| `value1`       | <pre>= [^\\}]<br/>\| '\\' quoted</pre>                         | Lm,<br/>\\n,<br/>\\}                    |
-| `value2`       | <pre>= [^:]<br/>\| '\\' quoted</pre>                           | Lm,<br/>\\n,<br/>\\:                    |
-| `char`         | <pre>= [^\\& \\- \\[ \\[ \\] \\\\ \\} \\{ [:Pat_WS:]]<br/>\| '\\' quoted</pre> | a, b, c, \\n                            |
-| `quoted`       | <pre>= 'u' (hex{4} \| bracketedHex)<br/>\| 'x' (hex{2} \| bracketedHex)<br/>\| 'U00' ('0' hex{5} \| '10' hex{4})<br/>\| 'N{' propName '}'<br/>\| [[\u0000-\U00010FFFF]-[uxUN]]</pre> | _**error** if lengths not exact_ |
-| `charName`     | <pre>= s [A-Za-z0-9] [-A-Za-z0-9_\x20]* s</pre>                | TIBETAN LETTER -A                       |
-| `bracketedHex` | <pre>= '{' s hexCodePoint (s hexCodePoint)* s '}'</pre>        | \{61 2019 62}                           |
-| `hexCodePoint` | <pre>= hex{1,5} \| '10' hex{4}</pre>                           |                                         |
-| `hex`          | <pre>= [0-9A-Fa-f]</pre>                                       |                                         |
-| `s`            | <pre>= [:Pattern_White_Space:]*</pre>                          | optional whitespace                     |
-
-Some constraints on UnicodeSet syntax are not captured by this EBNF. Notably, property names and values are restricted to those supported by the implementation, and have additional constraints imposed by [[UAX44](https://www.unicode.org/reports/tr41/#UAX44)]. In addition, quoted values that resolve to more than one code point are disallowed in ranges of the form `char '-' char`.
+| `root`         | <pre>= prop<br/>\| '\[' '^'? s '-'? s seq\* \[\\$ \\-\]? s '\]' <br/>\| var</pre> | \\p\{x=y\},<br/>[abc],<br/>$myset                     |
```

Comment:
To be clear about the handling of whitespace in the `root` nonterminal: ICU was the main inspiration for this, i.e., the previously UTS35-unsupported cases `[^-]`, `[^ -]`, `[ -]` *are* supported in ICU, and *are* supported in this new syntax.

One caveat: Per your suggestion I disallowed whitespace in front of the `^` for negation, even though ICU currently supports `[ ^]` (same as `[^]`).

Reply:
What you need to look at is the behavior with "free-spacing" mode (?x).
See https://gist.github.com/CMCDragonkai/6c933f4a7d713ef712145c5eb94a1816#:~:text=(%3Fx)%20(-,free%2Dspacing%20mode,-)

Reply:
Got it, thanks! I think we might even need to talk about `(?xx)`, because `(?x)` in PCRE still treats whitespace in character classes significantly - https://www.regular-expressions.info/freespacing.html

With `(?x)`: The behavior I described in the previous comment
With `(?xx)`: whitespace is ignored, for both `^` and `-`. `(?xx)[ ^ a- ]` matches everything except `a` and `-`.

Reply:
> With (?xx): whitespace is ignored, for both ^ and -. (?xx)[ ^ a- ] matches everything except a and -.

In that case we should keep the current ICU behaviour (`[ ^ a ]` = `[^a]`), since compatibility agrees with consistency with PCRE.

Reply:
This belongs here: While investigating properties, I noticed that ICU also ignores whitespace between `:` and `^` in e.g. `[:  ^ L :]`